pkg/api/dashboard.go

```diff
@@ -170,13 +172,12 @@ func (hs *HTTPServer) GetDashboard(c *models.ReqContext) response.Response {
 	// make sure db version is in sync with json model version
 	dash.Data.Set("version", dash.Version)
 
-	if hs.Cfg.IsPanelLibraryEnabled() {
-		// load library panels JSON for this dashboard
-		err = hs.LibraryPanelService.LoadLibraryPanelsForDashboard(c, dash)
-		if err != nil {
-			return response.Error(500, "Error while loading library panels", err)
-		}
+	// load library panels JSON for this dashboard
+	err = librarypanels.NewService(hs.SQLStore).LoadLibraryPanelsForDashboard(c, dash)
```

Comment:
I thought that this looked weird too, but I need some help to solve this. Have no knowledge about mutex in Go.

Reply:
@hugohaggmark I can give you some pointers (no pun intended).

Reply:
@hugohaggmark basically the old way, which was to have `LibraryPanelService` and `LibraryElementService` on the `HTTPServer` type is correct. I.e., to inject these services via our DI framework, which ensures that only one copy of each service is around. This will make eventual mutex usage safe in the services safe (even though they might not use mutexes now, that can easily change in the future).

Reply:
@aknuds1 the reason is that I need to call NewService instead so that I'm able to mock that in the tests?

Reply:
Just for context, @hugohaggmark and I discussed this further in chat and think we have a solution.